src/Couchbase.Lite.Shared/API/Database/Collection.cs

```diff
@@ -997,7 +997,7 @@ public override bool Equals(object obj)
                 return false;
             }
 
-            return IsValid
+            return ThreadSafety.DoLocked(() => IsValid)
```

Comment:
This is not solving a race, as other calculations are going to happen before the return.  If you want to eliminate the race you should do the entire calculation inside the locked part.